### PR TITLE
Reader: expanded photo cards - two heads are better than one

### PR DESCRIPTION
**Pull request by samouri**
a fork of: https://github.com/Automattic/wp-calypso/pull/10215

The solution attempted here is use the aspect ratio of the image to figure out the height the photo should be displayed at.
The math is: `(naturalHeight/naturalWidth) * actuallyAvailableWidth` --> height
TODO:
- [x] behavior as expected? @fraying 
- [x] too many hacks? @bluefuton 
- [x] tracks event added: `calypso_reader_photo_expanded`

### CODE REVIEW

**Comment by fraying** [comment]
Looking good!

Would it be possible to make the link title "Expand this photo" for the first click, and "View full post" for the second? The itch I'm trying to scratch is warning the users about what each click will do.

File: client/blocks/reader-post-card/photo.jsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		if ( this.state.isExpanded ) {
+			return;
+		}
+
+		// If the photo's not expanded, don't open full post yet
+		event.preventDefault();
+		this.setState( { isExpanded: true } );
+	}
+
+	// might need to debounce this
```
Comment (bluefuton) [inline client/blocks/reader-post-card/photo.jsx]: Can ditch this comment 👍

Comment (samouri) [inline client/blocks/reader-post-card/photo.jsx]: comment ditched

**Comment by samouri** [comment]
Conducting one final test then shipping:
- [ ] streams load properly
- [ ] cards expand and fire off stat
- [ ] zoom icon works correctly
- [ ] mysites loads

**Pull request merged**